reporter/reporter.go

```diff
@@ -76,13 +80,6 @@ func LogPercent(percent float32) Option {
 	}
 }
 
-// Custom RED metrics tags
-func RedMetricsCustomTagKeys(redMetricsCustomTagKeys []string) Option {
```

Comment:
@yzheqing Let's add this back. This is the setter end users use to specify which span tags they want to be promoted to the red metrics. And it's best not to change the input for backwards compat reasons. 

However, you can change the internal representation to a map. Just translate the provided slice to a map and set it internally.